employee-rostering-frontend/src/ui/components/OptionalInput.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { Switch, TextInput } from '@patternfly/react-core';
+
+export interface OptionalInputProps<T> {
+  defaultValue: T|null;
+  isValid: (value: string) => boolean;
+  valueMapper: (value: string) => T;
+  valueToString: (value: T) => string;
+  onChange: (value: T|null) => void;
+  label: string;
+}
+
+interface OptionalInputState<T> {
+  inputValue: T|null;
+  currentValue: T|null;
```

Comment:
The state property `currentValue` is unused. Reported by an ESLint rule [react/no-unused-state](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-state.md) that I'm trying out locally at the moment.